packages/slice-machine/lib/models/common/http/DefaultClient.ts

```diff
@@ -144,7 +144,7 @@ export default class DefaultClient {
         () => new Error(`Unable to parse profile: ${result.data}`)
       )(UserProfile.decode(result.data));
     } catch (e) {
-      // TODO: why not throw?
+      if (axios.isAxiosError(e)) throw e;
```

Comment:
I wouldn't throw, i don't like to throw if the function can return an error.
Here we need to choose
- throw and adapt when the function is called.
- return an error and adapt when it's called.

I don't think it's appropriate to have both.